lightning/src/chain/keysinterface.rs

```diff
@@ -1326,12 +1338,30 @@ impl SignerProvider for KeysManager {
 		InMemorySigner::read(&mut io::Cursor::new(reader))
 	}
 
-	fn get_destination_script(&self) -> Script {
-		self.destination_script.clone()
+	fn get_destination_script(&self, channel_keys_id: [u8; 32]) -> Script {
+		let derivation_idx = if self.new_style_derivation{
+			u64::from_be_bytes(channel_keys_id[0..8].try_into().unwrap())
+		} else {
+			3
+		};
+		match ExtendedPrivKey::new_master(Network::Testnet, &self.seed) {
+			Ok(master_key) => {
+				match master_key.ckd_priv(&self.secp_ctx, ChildNumber::from_hardened_idx(derivation_idx as u32).expect("key space exhausted")) {
```

Comment:
Is there a way to do it from u32? I looked at the `derive_channel_keys` as an example and saw that we get the byte slice using the method implemented for u64 and then do something like `(chan_id as 32) % (1<<31)` to avoid the failure when the top bit is set. When I tried to get the slice using the `from_be_bytes` method implemented for u32,  a bunch of tests start breaking.